Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -175,6 +180,11 @@ namespace AzToolsFramework
         AZ_Assert(IsInitialized(), "Tried to destroy an entity without initializing the Entity Ownership Service");
         AZ_Assert(m_entitiesRemovedCallback, "Callback function for DestroyEntityById has not been set.");
         OnEntityRemoved(entityId);
+        auto owningInstance = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
+        if (owningInstance.has_value())
```

Comment:
nit: We could update this to...

```c++

if (auto owningInstance = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
    owningInstance.has_value())
```

Just to limit the scope of `owningInstance`, also a guide comment here could be useful to indicate why this is required.